beforematch-security-and-privacy-assessment.md

```diff
@@ -0,0 +1,165 @@
+* In the ScrollToTextFragment case, text inside of a cross origin iframe will not be searched for or scrolled to. Therefore, no beforematch event will be emitted.
+* In the find-in-page case, content in cross origin iframes can be searched for and scrolled to. In this case, the cross origin iframe will get the beforematch event and scrolling will happen as normal, but the main frame won't know anything about it, which maintains the insulation between origins.
+
+### 2.14. How does this specification work in the context of a user agent's Private Browsing or "incognito" mode?
+
+This feature will work the same regardless of incognito mode being enabled or disabled.
+
+### 2.15. Does this specification have a "Security Considerations" and "Privacy Considerations" section?
+
+I included a Privacy Considerations section further down this document. I did not include a Security Considerations section because I do not believe this feature exposes any security vulnerabilities.
+
+### 2.16. Does this specification allow downgrading default security characteristics?
+
+No.
+
+### 2.17. What should this questionnaire have asked?
+
+I have no suggestions to improve this questionnaire.
+
+## Privacy Considerations
+
+### 1. What privacy attacks have been considered?
+
+The beforematch event could be used to guess what the user is searching for.
+* In the ElementFragment case, there are no new vulnerabilities. Any information possibly exposed by beforematch is already exposed by `window.location.hash`, the `hashchange` event, and `document.getElementById`. The following code snippet's `elementFromFragment` is the same element that `beforematch` will fire on when an element is scrolled to on hash change:
+```javascript
+window.addEventListener('hashchange', () => {
+  const elementFromFragment = document.getElementById(window.location.hash);
+});
+```
+* In the ScrollToTextFragment case, the beforematch event will hint to what the user was searching for by firing on the block level element containing the text which was searched for. However, the :target pseudo selector which is already added by ScrollToTextFragment is also applied to the same element that beforematch would fire on, so the beforematch event will not expose anything new in this case.
+* In the find-in-page case, the same information will be exposed as the ScrollToTextFragment case. However, if the user is incrementally searching by typing in a search character by character, it will result in repeated beforematch events being fired. It may be possible to reconstruct what the user is searching for in this case by having elements in the page which match every next possible character the user could search for, and then when a match occurs, changing the text of all of the elements to contain every next possible match, and so on. Here is an example page which conveys this idea. I couldn't actually get it to work properly, but with some more work I think it might:
```

Comment:
```suggestion
* In the find-in-page case, the same information will be exposed as the ScrollToTextFragment case. However, if the user is incrementally searching by typing in a search dialog character by character, it will result in repeated beforematch events being fired. It may be possible to reconstruct what the user is searching for in this case by having elements in the page which match every next possible character the user could search for, and then when a match occurs, changing the text of all of the elements to contain every next possible match, and so on. Here is an example page which conveys this idea. I couldn't actually get it to work properly, but with some more work I think it might:
```